src/app/ng2wig/ng2wig.component.ts

```diff
@@ -0,0 +1,421 @@
+import {Component, Input, OnInit, OnChanges, ViewEncapsulation, forwardRef, SimpleChanges} from '@angular/core';
+import {NG_VALUE_ACCESSOR, ControlValueAccessor} from '@angular/forms';
+import {Ng2WigToolbarService} from './ng2wig-toolbar.service';
+
+@Component({
+  selector: 'ng2wig',
+  template: `
```

Comment:
It would be better to have template and styles in their separate files. It will be more readable and conform to the Angular Style Guide.